### PR TITLE
New package: ocs-url-3.1.0

### PR DESCRIPTION
**Pull request by linuxpotion**

### CODE REVIEW

**Comment by ndowens** [comment]
No need to put template in comment :)

File: srcpkgs/ocs-url/template [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'ocs-url'
+pkgname=ocs-url
+version=3.1.0
+revision=1
+#archs="x86_64"
```
Comment (ndowens) [inline srcpkgs/ocs-url/template]: Remove commented line

File: srcpkgs/ocs-url/template [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+wrksrc=$pkgname-release-$version
+build_style=qmake
+hostmakedepends="qt5-qmake xdg-utils git"
+makedepends="qt5-devel qt5-svg-devel qt5-declarative-devel qt5-quickcontrols2-devel"
+short_desc="Install helper program for items served via (ocs://)"
+maintainer="Guilherme Machado <machadofguilherme@gmail.com>"
+license="GPL-3.0"
```
Comment (ndowens) [inline srcpkgs/ocs-url/template]: Needs to be either GPL-3.0-only or GPL-3.0-or-later

File: srcpkgs/ocs-url/template [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+homepage="https://www.opencode.net/akiraohgaki/$pkgname"
```
Comment (ndowens) [inline srcpkgs/ocs-url/template]: Personally prefer homepage to not have variables, makes it easier if someone wants to copy and paste the URL to visit upstream's site.

File: srcpkgs/ocs-url/template [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+distfiles="https://www.opencode.net/akiraohgaki/ocs-url/-/archive/release-$version/$pkgname-release-$version.tar.gz"
+checksum=d8c67920359c36eb0004221e5b5695fd04cbc9e93811c372837bdb2a8c9cbcc8
+
+pre_configure() {
+./scripts/prepare
+}
+
+do_configure() {
+qmake PREFIX="/usr"
```
Comment (ndowens) [inline srcpkgs/ocs-url/template]: Remove do_configure block, it is not need.

File: srcpkgs/ocs-url/template [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+do_install() {
+make INSTALL_ROOT="${PKGDESTDIR}" install
```
Comment (ndowens) [inline srcpkgs/ocs-url/template]: Remove do_install() block, works fine without manual install.

File: srcpkgs/ocs-url/template [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+makedepends="qt5-devel qt5-svg-devel qt5-declarative-devel qt5-quickcontrols2-devel"
+short_desc="Install helper program for items served via (ocs://)"
+maintainer="Guilherme Machado <machadofguilherme@gmail.com>"
+license="GPL-3.0"
+homepage="https://www.opencode.net/akiraohgaki/$pkgname"
+distfiles="https://www.opencode.net/akiraohgaki/ocs-url/-/archive/release-$version/$pkgname-release-$version.tar.gz"
+checksum=d8c67920359c36eb0004221e5b5695fd04cbc9e93811c372837bdb2a8c9cbcc8
+
+pre_configure() {
+./scripts/prepare
```
Comment (ndowens) [inline srcpkgs/ocs-url/template]: Also need to indent

File: srcpkgs/ocs-url/template (hunk 1 above)
Comment (linuxpotion) [reply srcpkgs/ocs-url/template]: > Remove commented line

Commented line removed. =]

**Comment by ndowens** [comment]
Working on the cross issue and also can remove git when I get done

File: srcpkgs/ocs-url/template (hunk 2 above)
Comment (linuxpotion) [reply srcpkgs/ocs-url/template]: > Needs to be either GPL-3.0-only or GPL-3.0-or-later

License fixed. =]

File: srcpkgs/ocs-url/template (hunk 3 above)
Comment (linuxpotion) [reply srcpkgs/ocs-url/template]: > Personally prefer homepage to not have variables, makes it easier if someone wants to copy and paste the URL to visit upstream's site.

Homepage fixed. =]

**Comment by ndowens** [comment]
Ok here is my template, of course you need to fix the License info:
https://termbin.com/vir5x

File: srcpkgs/ocs-url/template (hunk 4 above)
Comment (linuxpotion) [reply srcpkgs/ocs-url/template]: > Remove do_configure block, it is not need.

Removed. =]

File: srcpkgs/ocs-url/template (hunk 5 above)
Comment (linuxpotion) [reply srcpkgs/ocs-url/template]: > Remove do_install() block, works fine without manual install.

Removed. =]

**Comment by ndowens** [comment]
You can look at my template I did, and use it remove the need of git and
fixes cross-build issue as well

File: srcpkgs/ocs-url/template (hunk 6 above)
Comment (linuxpotion) [reply srcpkgs/ocs-url/template]: > Also need to indent

Man, I added your suggestions. Take a look when you can. xD

Comment (ndowens) [reply srcpkgs/ocs-url/template]: > > Also need to indent
> 
> Man, I added your suggestions. Take a look when you can. xD

Look at the distfiles, I have another source there, it is the one that it is downloading another source it needs.

File: srcpkgs/ocs-url/template [hunk 7]
```diff
@@ -10,8 +10,8 @@ short_desc="Install helper program for items served via (ocs://)"
 maintainer="Guilherme Machado <machadofguilherme@gmail.com>"
 license="GPL-3.0-or-later"
 homepage="https://www.opencode.net/akiraohgaki/$pkgname"
-distfiles="https://www.opencode.net/akiraohgaki/ocs-url/-/archive/release-$version/$pkgname-release-$version.tar.gz"
-checksum="d8c67920359c36eb0004221e5b5695fd04cbc9e93811c372837bdb2a8c9cbcc8"
+distfiles="https://www.opencode.net/akiraohgaki/ocs-url/-/archive/release-$version/$pkgname-release-$version.tar.gz https://github.com/akiraohgaki/qtil/archive/release-0.4.0.tar.gz"
```
Comment (ndowens) [inline srcpkgs/ocs-url/template]: Needs to look like how I did it on a new line

File: srcpkgs/ocs-url/template (hunk 6 above)
Comment (linuxpotion) [reply srcpkgs/ocs-url/template]: > > > Also need to indent
> > 
> > 
> > Man, I added your suggestions. Take a look when you can. xD
> 
> Look at the distfiles, I have another source there, it is the one that it is downloading another source it needs.

I had not seen. I tested the template with the modifications you submitted. XBPS-SRC created the package without problems without git between dependencies. I submitted the template changes again for review.

**Comment by Chocimier** [comment]
Included in #23679

**Pull request closed without merging**